.github/workflows/python_build.yml

```diff
@@ -36,29 +36,26 @@ jobs:
     name: Python Build (Python 3.7 PyArrow 7.0.0)
     runs-on: ubuntu-latest
     # use the same environment we have for python release
-    container: quay.io/pypa/manylinux2010_x86_64:2022-03-14-b2cd80b
+    container: quay.io/pypa/manylinux2014_x86_64:latest
```

Comment:
should we pin to a specific image tag instead of latest for better reproducibility?